astropy/coordinates/representation.py

```diff
@@ -42,8 +42,12 @@ def __init__(cls, name, bases, dct):
         if cls.__name__ == 'BaseRepresentation':
             return
 
-        REPRESENTATION_CLASSES[cls.get_name()] = cls
+        repr_name = cls.get_name()
 
+        if repr_name in REPRESENTATION_CLASSES:
+            raise ValueError("Representation class {0} already defined".format(repr_name))
+
+        REPRESENTATION_CLASSES[repr_name] = cls
```

Comment:
@eteq - any objections to this? This will not allow two representations named the same to be defined - that's ok, right? Or would a warning be more appropriate?